### PR TITLE
fix: move oidc scope logic to oidc provider

### PR DESCRIPTION
**Pull request by tuunit**
## Description and Motivation
The OIDC provider shouldn't be treated as anything specially where as all the other providers have there internal scope logic defined in the specific provider files the scope logic for oidc and in extension the keycloak-oidc provider is located in the providers.go.

To align the OIDC and Keycloak OIDC provider I moved the logic to the oidc.go provider file.

## Checklist:

- [X] My change requires a change to the documentation or CHANGELOG.
- [X] I have updated the documentation/CHANGELOG accordingly.
- [X] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

**Comment by tuunit** [comment]
@Kyserbyte I tested your issue with the scope missing for the KeycloakOIDC provider and couldn't produce it. As the KeycloakOIDC provider is a extension of the OIDC provider it should have the same default scope of "openid email profile". Nevertheless, I think it is a good idea to move this logic from the generic provider.go to the OIDC provider.

File: providers/providers.go [hunk 1]
```diff
@@ -152,17 +152,6 @@ func newProviderDataFromConfig(providerConfig options.Provider) (*ProviderData,
...
 
-	if providerConfig.Type == "oidc" && p.Scope == "" {
-		p.Scope = "openid email profile"
-
-		if len(providerConfig.AllowedGroups) > 0 {
-			p.Scope += " groups"
-		}
-	}
-	if providerConfig.OIDCConfig.UserIDClaim == "" {
-		providerConfig.OIDCConfig.UserIDClaim = "email"
```
Comment (tuunit) [inline providers/providers.go:163]: This is dead code. As it is not a reference and not used after this line.

File: CHANGELOG.md [hunk 2]
```diff
@@ -16,6 +16,7 @@
 - [#1927](https://github.com/oauth2-proxy/oauth2-proxy/pull/1927) Fix default scope settings for none oidc providers
 - [#1951](https://github.com/oauth2-proxy/oauth2-proxy/pull/1951) Fix validate URL, check if query string marker (?) or separator (&) needs to be appended (@miguelborges99)
 - [#1920](https://github.com/oauth2-proxy/oauth2-proxy/pull/1920) Make sure emailClaim is not overriden if userIDClaim is not set
+- [#1989](https://github.com/oauth2-proxy/oauth2-proxy/pull/1989) Fix default scope for keycloak-oidc provider
```
Comment (tuunit) [inline CHANGELOG.md:19]: @tuunit reminder to myself to change this changeling to a better description of this change

**Comment by tuunit** [comment]
@JoelSpeed please reopen :)

**Comment by JoelSpeed** [comment]
The branch has since been force pushed, so I can't currently reopen right now

**Pull request closed without merging**